### ISSUE TITLE
CI

### ISSUE THREAD

**Issue by thorsteneckel** [thorsteneckel/threema#8]
For testing failing requests in case of unsuccessful HTTP Public Key Pinning there is currently a private request-bin in use to prevent leaking credentials:

https://github.com/thorsteneckel/threema/blob/55c406869528c8f31e17ed44c692745daa49062f/spec/threema/client_spec.rb#L56

https://github.com/thorsteneckel/threema/blob/55c406869528c8f31e17ed44c692745daa49062f/spec/threema/client_spec.rb#L61

This is necessary since (currently) the API credentials get added to the requests. However this is not a proper way. There should be a dedicated method for testing this.

**Comment by rugk** [comment]
Maybe define them [in Travis](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings) or [encrypt them](https://docs.travis-ci.com/user/encryption-keys/).

**Comment by thorsteneckel** [comment]
What I mean is that if I disable the pinning (or whatever I'm currently doing 😇) I have to request a https domain and the credentials will get send to that. That's ok when working with fake credentials as I currently do but when testing locally with real credentials this is not what someone wants to do.

However your proposal might be a solution for issue #2 .

**Comment by rugk** [comment]
> whatever I'm currently doing

Ah, all right… maybe we need another issue for that as you said in the comments already.

**Comment by thorsteneckel** [comment]
Yes! I'll have to read me into the topic first. My "blueprint" was the python client which has the same feature. Idk if you have influence on this but it uses a deprecated -> [via SHA1](https://github.com/lgrahl/threema-msgapi-sdk-python/blob/bed367b9a8775aa1ea2d1972fd1a6cf85798238c/threema/gateway/__init__.py#L86). I had no time yet to create an issue.

**Comment by rugk** [comment]
Yes, you obviously also had no time to look for an existing issue… :wink: 
https://github.com/lgrahl/threema-msgapi-sdk-python/issues/17

So yes, and I explained there in detail why pinning the certificate is bad. You should really have a look into what HPKP is, which is a good technique for pinning (except that in your case you'd pin the hash statically instead of using the cert hash.
